blaze/server/serialization.py

```diff
@@ -2,13 +2,65 @@
 from functools import partial
 import json as json_module
 
+from datashape.predicates import iscollection, istabular, isscalar
+from odo import odo
+import pandas as pd
+from pandas.compat import BytesIO
+from pandas.io.packers import (
+    unpack as msgpack_unpack,
+    decode as pd_msgpack_object_hook,
+    encode as pd_msgpack_default,
+)
 import pandas.msgpack as msgpack_module
+from toolz import identity
 
-from ..compatibility import pickle as pickle_module, unicode
+from ..compatibility import pickle as pickle_module, unicode, PY2
+from ..interactive import coerce_scalar
 from ..utils import json_dumps, object_hook
 
 
-SerializationFormat = namedtuple('SerializationFormat', 'name loads dumps')
+_serializationformat = namedtuple(
+    'SerializationFormat', (
+        'name',
+        'loads',
+        'dumps',
+        'data_loads',
+        'data_dumps',
+        'materialize',
+    ),
+)
+
+
+class SerializationFormat(_serializationformat):
+    def __new__(cls,
+                name,
+                loads,
+                dumps,
+                data_loads=None,
+                data_dumps=None,
+                materialize=None):
```

Comment:
If we don't need slots (and don't need to subclass from a `namedtuple`) that's great by me -- my comment was more of a question if you wanted to set an empty slots in conjunction with the subclass, which is now a moot point :)